src/modules/hip/kernel/crop_mirror_normalize.hpp

```diff
@@ -65,7 +65,7 @@ __global__ void crop_mirror_normalize_pkd_tensor(T *srcPtr,
         if((id_z == 0) && (id_y == 0) && (id_x + 8) > roiTensorPtrSrc[id_z].xywhROI.roiWidth)
         {
             srcIdx = (id_z * srcStridesNH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + roiTensorPtrSrc[id_z].xywhROI.xy.x * 3;
-            dstIdx -= (roiTensorPtrSrc[id_z].xywhROI.roiWidth - id_x) * 3;
+            dstIdx -= (id_x + 8 - roiTensorPtrSrc[id_z].xywhROI.roiWidth) * 3;
```

Comment:
Hi Rajy,
This condition is needed only for the 1st row of the 1st image in the batch 
- when image width is not a multiple of 8
- when mirror is set to 1

for a 50x50 input image(image width is not multiple of 8) when mirror = 1
for the first 8 pixels stored in output row we will load the last 8 values in input row 
do the mirror, normalize and store in output

This will happen till 48 values in a row. For last 2 locations in output - 48 49
we are loading 8 pixels from input from locations 0 1 2 3 4 5 6 7
srcIdx = (id_z * srcStridesNH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + roiTensorPtrSrc[id_z].xywhROI.xy.x * 3;
mirrored as 7 6 5 4 3 2 1 0
The destination location (dstIdx) at this case is 48

To use the 8 pixel store here as well, we are updating the destination pointer to decrement by below factor
`dstIdx -= (id_x + 8 - roiTensorPtrSrc[id_z].xywhROI.roiWidth)`

In this case it will be decremented by (48 + 8 - 50) = 6
Now dstIdx will be present at 42 and values at 7 6 5 4 3 2 1 0 locations will be stored in output
For only this case we are overwriting already existed values in destination at 42 43 44 45 46 47 locations to use the 8 pixel store